geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -216,3 +215,15 @@ class GeoMesaSparkKryoRegistrator extends KryoRegistrator {
     SimpleFeatureSerializers.simpleFeatureImpls.foreach(kryo.register(_, serializer, kryo.getNextRegistrationId))
   }
 }
+
+object GeoMesaSparkKryoRegistrator {
+  val typeCache: ConcurrentHashMap[String, SimpleFeatureType] = new ConcurrentHashMap[String, SimpleFeatureType]()
+
+  def putType(sft: SimpleFeatureType): Unit = {
+    GeoMesaSparkKryoRegistrator.typeCache.putIfAbsent(sft.getTypeName, sft)
```

Comment:
Wondering if there's a case where a type would need to be updated. I've noticed in Jupyter that static variables tend to hang around even after a kernel restart